### DISCUSSION TITLE
how exactly the downloader works

### DISCUSSION THREAD

**Discussion by purplegrapeZz** [Dineshkarthik/telegram_media_downloader#169] [Q&A] [answered]
Hi all, 
I've tried and it works fine, great!
I have read the code but I can not figure out how the downloader exactly saves media.
I know asyncio but I did not find any code like 'save()'.
So how exactly the downloader saves files?

**Comment by purplegrapeZz** [comment, marked as answer]
hi all,
I think it is
# L191
download_path = await client.download_media(
                            message, file_name=file_name
                        )

I believe I can find the answer in pyrogram docs.

pyrogram is good,can work in all channels.
I have tried telethon earlier on but it can not downloads in channels that are not allowed to download.

**Reply by Dineshkarthik** [reply]
Yes, you are correct telegram-media-downloader uses `Pyrogram` to interact with telegram APIs, #L191 is where the pyrogram method to download media is being invoked.
